tempest/api/workloadmgr/restore/test_network_restore.py

```diff
@@ -0,0 +1,186 @@
+                subnet = self.subnets_client.create_subnet(**subnetconfig)
+                subnets[subnet['subnet']['name']] = subnet['subnet']['id']
+            else:
+                net = self.networks_client.create_network(**{'name':"Private-{}".format(x), 'admin_state_up':'False', 'shared':'True'})
+                nets[net['network']['name']] = net['network']['id']
+
+        for x in range(1,6):
+            if x != 4:
+                router = self.network_client.create_router(**{'name':"Router-{}".format(x)})
+            else:
+                router = self.network_client.create_router(**{'name':"Router-{}".format(x), 'admin_state_up':'False'})
+            routers[router['router']['name']] = router['router']['id']
+
+        networkslist = self.networks_client.list_networks()['networks']
+        self.network_client.add_router_interface_with_subnet_id(routers['Router-1'], subnets['PS-1'])
+        self.network_client.add_router_interface_with_subnet_id(routers['Router-1'], subnets['PS-2'])
+        self.network_client.add_router_interface_with_subnet_id(routers['Router-3'], subnets['PS-3'])
+        self.network_client.add_router_interface_with_subnet_id(routers['Router-2'], subnets['PS-4'])
+        portid1 = self.network_client.create_port(**{'network_id':nets['Private-2'], 'fixed_ips': [{'ip_address':'10.10.2.4'}]})['port']['id']
+        self.network_client.add_router_interface_with_port_id(routers['Router-2'], portid1)
+        portid2 = self.network_client.create_port(**{'network_id':nets['Private-2'], 'fixed_ips': [{'ip_address':'10.10.2.5'}]})['port']['id']
+        portid3 = self.network_client.create_port(**{'network_id':nets['Private-2'], 'fixed_ips': [{'ip_address':'10.10.2.6'}]})['port']['id']
+        self.network_client.add_router_interface_with_port_id(routers['Router-4'], portid2)
+        self.network_client.add_router_interface_with_port_id(routers['Router-5'], portid3)
+        self.network_client.add_router_interface_with_subnet_id(routers['Router-4'], subnets['PS-5'])
+        portid4 = self.network_client.create_port(**{'network_id':nets['Private-5'], 'fixed_ips': [{'ip_address':'10.10.5.3'}]})['port']['id']
+        self.network_client.add_router_interface_with_port_id(routers['Router-5'], portid4)
+        
+    def get_topology_details(self):
+            networkslist = self.networks_client.list_networks()['networks']
+            nws = [x['id'] for x in networkslist]
+            nt= [{str(i):str(j) for i,j in x.items() if i not in ('network_id', 'subnets', 'created_at', 'updated_at', 'id')} for x in networkslist]
+            networks = {}
+            for each in nt:
+                networks[each['name']] = each
+
+            sbnt = self.subnets_client.list_subnets()['subnets']
+            sbnts = [{str(i):str(j) for i,j in x.items() if i not in ('network_id', 'created_at', 'updated_at', 'id')} for x in sbnt]
+            subnets = {}
+            for each in sbnts:

```

Comment:
Please use variable names which convey the details appropriately. Rename this variable 'each' .